python_modules/dagster/dagster/core/decorators.py

```diff
@@ -28,15 +40,28 @@ def has_context(self):
 
 
 class _Solid:
-    def __init__(self, name=None, inputs=None, output=None, description=None):
+    def __init__(
+        self,
+        name=None,
+        inputs=None,
+        outputs=None,
+        output=None,
+        description=None,
+        config_def=None,
+    ):
         self.name = check.opt_str_param(name, 'name')
         self.inputs = check.opt_list_param(inputs, 'inputs', InputDefinition)
 
-        check.opt_inst_param(output, 'output', OutputDefinition)
-        if not output:
-            output = OutputDefinition()
-        self.output = output
+        if output is not None and outputs is None:
```

Comment:
Let's make this a very explicit error message. Detect when both are None. Do specific error message. Detect when both are not none. Do specific error message.